cmd/scollector/collectors/procstats_linux.go

```diff
@@ -36,171 +28,165 @@ var CPU_FIELDS = []string{
+	Add(&md, "linux.loadavg.5_min", sys.Loads[1], nil, metadata.Gauge, metadata.Load, "")
+	Add(&md, "linux.loadavg.15_min", sys.Loads[2], nil, metadata.Gauge, metadata.Load, "")
+	Add(&md, "linux.loadavg.total_threads", sys.Procs, nil, metadata.Gauge, metadata.Process, "")
+	if sys.Totalram != 0 {
+		Add(&md, osMemPctFree, (sys.Freeram)/(sys.Totalram)*100, nil, metadata.Gauge, metadata.Pct, osMemFreeDesc)
 	}
 	if err := readLine("/proc/vmstat", func(s string) error {
-		m := vmstatRE.FindStringSubmatch(s)
+		m := strings.Split(s, " ")
 		if m == nil {
 			return nil
 		}
-		switch m[1] {
-		case "pgpgin", "pgpgout", "pswpin", "pswpout", "pgfault", "pgmajfault":
-			mio := inoutRE.FindStringSubmatch(m[1])
-			if mio != nil {
-				Add(&md, "linux.mem."+mio[1], m[2], opentsdb.TagSet{"direction": mio[2]}, metadata.Counter, metadata.Page, "")
-			} else {
-				Add(&md, "linux.mem."+m[1], m[2], nil, metadata.Counter, metadata.Page, "")
+		switch m[0] {
+		case "pgpgin", "pgpgout", "pswpin", "pswpout":
+			switch {
+			case strings.HasSuffix(m[0], "in"):
+				Add(&md, "linux.mem."+strings.TrimSuffix(m[0], "in"), m[1], opentsdb.TagSet{"direction": "in"}, metadata.Counter, metadata.Page, "")
+			case strings.HasSuffix(m[0], "out"):
+				Add(&md, "linux.mem."+strings.TrimSuffix(m[0], "out"), m[1], opentsdb.TagSet{"direction": "out"}, metadata.Counter, metadata.Page, "")
 			}
+		case "pgfault", "pgmajfault":
+			Add(&md, "linux.mem."+m[0], m[1], nil, metadata.Counter, metadata.Page, "")
 		default:
-			Add(&md, "linux.mem."+m[1], m[2], nil, metadata.Counter, metadata.None, "")
+			Add(&md, "linux.mem."+m[0], m[1], nil, metadata.Counter, metadata.None, "")
 		}
 		return nil
 	}); err != nil {
+		slog.Errorln(err)
 		Error = err
 	}
 	num_cores := 0
-	var t_util float64
-	cpu_stat_desc := map[string]string{
-		"user":       "Normal processes executing in user mode.",
-		"nice":       "Niced processes executing in user mode.",
-		"system":     "Processes executing in kernel mode.",
-		"idle":       "Twiddling thumbs.",
-		"iowait":     "Waiting for I/O to complete.",
-		"irq":        "Servicing interrupts.",
-		"softirq":    "Servicing soft irqs.",
-		"steal":      "Involuntary wait.",
-		"guest":      "Running a guest vm.",
-		"guest_nice": "Running a niced guest vm.",
-	}
+	var t_util int
 	if err := readLine("/proc/stat", func(s string) error {
-		m := statRE.FindStringSubmatch(s)
+		m := strings.Fields(s)
 		if m == nil {
 			return nil
 		}
-		if strings.HasPrefix(m[1], "cpu") {
-			metric_percpu := ""
-			tag_cpu := ""
-			cpu_m := statCPURE.FindStringSubmatch(m[1])
-			if cpu_m != nil {
-				num_cores += 1
-				metric_percpu = ".percpu"
-				tag_cpu = cpu_m[1]
+		switch {
+		case strings.HasPrefix(m[0], "cpu"):
+			tag_cpu := strings.TrimPrefix(m[0], "cpu")
+			if tag_cpu != "" {
+				num_cores++
 			}
-			fields := strings.Fields(m[2])
-			for i, value := range fields {
-				if i >= len(CPU_FIELDS) {
+			for i, value := range m[1:] {
```

Comment:
check len here to. Basically m will never be nil now that split is being uses, so the nil check is pointless, and need to check len.